main/core/messagebroker.ts

```diff
@@ -36,6 +38,8 @@ export function messagebroker<T = any>(): IMessageBroker<T> {
 export class MessageBroker<T = any> implements IMessageBroker<T> {
     private channelLookup: ChannelModelLookup<T> = {};
     private messagePublisher = new Subject<IMessage<any>>();
+    private adapters: IMessageBrokerAdapter<T>[] = [];
```

Comment:
this might be better as a Record:

`Record<string, IMessageBrokerAdaptor>`

IMHO we don't really need an array as the order is not important. A `Record` will make it a bit neater removing adaptors. Any other access we can refer to all adaptors with `Object.values(this.adaptors)`